api/v1alpha1/instaslice_types.go

```diff
@@ -44,54 +58,103 @@ const (
 
 // Define the struct for allocation details
 type AllocationDetails struct {
-	Profile  string `json:"profile"`
-	Start    uint32 `json:"start"`
-	Size     uint32 `json:"size"`
-	PodUUID  string `json:"podUUID"`
-	GPUUUID  string `json:"gpuUUID"`
+	// profile requested by user workload
+	// +required
+	Profile string `json:"profile"`
+	// start position of a profile on a gpu
+	// +required
+	Start uint32 `json:"start"`
+	// size of profile that begins from start position
+	// +required
+	Size uint32 `json:"size"`
+	// podUUID represents uuid of user workload
+	// +required
+	PodUUID string `json:"podUUID"`
+	// gpuUUID represents gpu uuid of selected gpu
+	// +required
+	GPUUUID string `json:"gpuUUID"`
+	// nodename represents name of the selected node
+	// +required
 	Nodename string `json:"nodename"`
+	// allocationStatus represents status of allocation
 	// +kubebuilder:validation:Enum:=deleted;deleting;ungated;creating;created
-	Allocationstatus   AllocationStatus `json:"allocationStatus"`
-	Resourceidentifier string           `json:"resourceIdentifier"`
-	Namespace          string           `json:"namespace"`
-	PodName            string           `json:"podName"`
-	Cpu                int64            `json:"cpu"`
-	Memory             int64            `json:"memory"`
+	// +required
+	Allocationstatus AllocationStatus `json:"allocationStatus"`
+	// resourceIdentifier represents uuid used for creating configmap resource
+	// +required
+	Resourceidentifier string `json:"resourceIdentifier"`
+	// namespace represents namespace of user workload
+	// +required
+	Namespace string `json:"namespace"`
+	// podName represents name of the user workload pod
+	// +required
+	PodName string `json:"podName"`
+	// cpu represents amount of cpu requested by user workload
+	// +required
+	Cpu int64 `json:"cpu"`
+	// memory represents amount of memory requested by user workload
+	// +required
+	Memory int64 `json:"memory"`
 }
 
 // InstasliceSpec defines the desired state of Instaslice
 type InstasliceSpec struct {
-	MigGPUUUID         map[string]string            `json:"MigGPUUUID,omitempty"`
```

Comment:
Why go from omitEmpty to required?

Reply:
MIG GPU UUID is used to walk migs and we don't neet it empty in CR